### PR TITLE
IntersectsBox()

### PR DESCRIPTION
**Pull request by vedhant**
implements #44 
This PR depends on #48 .

### CODE REVIEW

File: src/p5xr/core/raycasting.js [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+  let sphereToRayOrigin = p5.Vector.sub(ray.origin, sphereCenter);
+  let b = 2 * p5.Vector.dot(ray.direction, sphereToRayOrigin);
+  let c = p5.Vector.mag(sphereToRayOrigin) * p5.Vector.mag(sphereToRayOrigin) - radius * radius;
+
+  let det = b * b - 4 * c;
+
+  return det >= 0;
+};
+
+p5.prototype.intersectsBox = function() {
```
Comment (stalgiag) [inline src/p5xr/core/raycasting.js]: Moving [a discussion ](https://github.com/stalgiag/p5.xr/pull/48/files#r305584870)to this PR:

Main issue is how to setup the parameters for `intersectsBox`. All of the following should be possible:
```
intersectsBox(10) //checks for intersection between a ray originating at screen 0, 0 and a box with dimensions 10, 10, 10
intersectsBox(10, 20, 10, 0, 0) //checks for intersection btwn a ray originating at screen 0, 0 and a box with dimensions 10, 20, 10 -- 
```
Note that the screenX and screenY must be specified in the last example. This is unfortunate but straightforward for the time being.

Additionally these should work
```
intersectsBox(10, ray) 
intersectsBox(10, 20, 10, ray)

Comment (vedhant) [reply src/p5xr/core/raycasting.js]: I have added the default values.

**Comment by vedhant** [comment]
@stalgiag This PR is now ready to be merged.

**Comment by stalgiag** [comment]
@vedhant can you rebase this branch with current `master` so that the diffs read correctly? The changes from #48 are coming through here.

**Pull request merged**